tests/test_model_sector.py

```diff
@@ -0,0 +1,37 @@
+""" Various tests for ModelSector and its interaction with TimingModel
+"""
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import os
+import numpy as np
+import astropy.units as u
+from copy import deepcopy
+
+from pint.models.timing_model import (
+    TimingModel,
+    Component,
+)
+from pint.models.model_sector import (
+    ModelSector,
+    DelaySector,
+)
+
+
+class TestSector:
```

Comment:
It seems worth checking that the code correctly rejects sectors that share "methods", that all listed methods in existing sectors actually exist, and that the "external_sector_map" actually does what it's supposed to.